debug_toolbar/panels/staticfiles.py

```diff
@@ -88,12 +97,25 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.num_found = 0
         self.used_paths = []
+        self.request_id = str(uuid.uuid4())
+
+    def _store_static_files_signal_handler(self, sender, staticfile, **kwargs):
+        with contextlib.suppress(LookupError):
+            # Only record the static file if the request_id matches the one
+            # that was used to create the panel.
+            # as sender of the signal and this handler will have multiple
+            # concurrent connections and we want to avoid storing of same
+            # staticfile from other connections as well.
+            if request_id_context_var.get() == self.request_id:
+                used_static_files.get().append(staticfile)
 
     def enable_instrumentation(self):
         storage.staticfiles_storage = DebugConfiguredStorage()
+        record_static_file_signal.connect(self._store_static_files_signal_handler)
+        request_id_context_var.set(self.request_id)
```

Comment:
We'll want to capture the [returned token](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar.set) here and then use it to [reset](https://docs.python.org/3/library/contextvars.html#contextvars.ContextVar.reset) the `ContextVar`